src/entities/Attendance.ts

```diff
@@ -10,21 +10,18 @@ import { Event } from './Event';
  */
 @Entity()
 export class Attendance {
-  @PrimaryGeneratedColumn()
-  id: number;
-
-  @ManyToOne(() => AppUser)
+  @ManyToOne(() => AppUser, { primary: true })
   attendee: AppUser;
 
   // Officer who checked off attendee
   @ManyToOne(() => AppUser)
   officer: AppUser;
 
-  @ManyToOne(() => Event)
+  @ManyToOne(() => Event, { primary: true })
   event: Event;
 
   // num_hours
-  @Column('float')
+  @Column('float', { nullable: true })
```

Comment:
out of scope but maybe duration can be a computed value of end time - start time? ik typeorm can autogenerate a creation time column. food for thought

Reply:
and tbf this is mb - i just jammed duration in there without much thought

Reply:
yeah that's what duration should be